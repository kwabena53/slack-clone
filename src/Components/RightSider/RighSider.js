
import { CloseIcon, HeadphoneIcon, MenuIcon, MessageIcon, VideoIcon } from "../Icons/SlackCloneIcons"
import "./RightSider.css"




const RightSider = ({closeSider})=>{
    return(
        <div className="rightSider">
            <div className="contentHeader h-line">
                <h3 className="slack-h3">Profile</h3>
                <div onClick={()=>closeSider(false)} className="sl-icon">
                    <CloseIcon width="2em" height="2em" color="black" />
                </div>
            </div>
            <div className="rightsider-content">
                <div className="profile-details">

              
                <div>
                    <img className="profile-pic" src="https://randomuser.me/api/portraits/women/9.jpg" />
                </div>
                <div>
                    <div className="user-name">
                        <h3 className="slack-h3">Jenn Rode</h3> 
                        </div>
                    <div className="online-status"></div>
                </div>
                <div className="profile-action-icons">
                    <div className="profile-icon-section">
                        <div className="round-circle">
                            <MessageIcon width="1em" height="1em" color="black"/>
                        </div>
                        <div className="profile-icon-name">Message</div>
                    </div>
                    <div className="profile-icon-section">
                        <div className="round-circle">
                        <HeadphoneIcon width="1em" height="1em" color="black"/>
                        </div>
                        <div className="profile-icon-name">Huddle</div>
                    </div>
                    <div className="profile-icon-section">
                        <div className="round-circle">
                        <MenuIcon width="1em" height="1em" color="black"/>
                        </div>
                        <div className="profile-icon-name">More</div>
                    </div>
                </div>
                </div>
                <div>
                <div className="personal-section">
                    <div className="details-title">
                        Display name
                    </div>
                    <div className="details-res">
                        jbsarfo
                    </div>
                </div>
                <div>
                    <div className="details-title">
                        Local time
                    </div>
                    <div className="details-res">
                        6:09 PM
                    </div>
                </div>
                <div>
                    <div className="details-title">
                        Email address
                    </div>
                    <div className="details-res">
                        jbsarfo@asu.edu
                    </div>
                </div>
                <div>
                    <div className="details-title">
                    Shared Workspaces
                    </div>
                    <div className="details-res">
                    EdPlus at ASU
                    </div>
                </div>
                </div>
                
            </div>
        </div>
    )
}

export default RightSider