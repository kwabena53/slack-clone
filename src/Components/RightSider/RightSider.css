



.rightSider{
    width: 120%;
    border-left: 1px solid rgb(0, 0, 0, 0.1);
}

.rightsider-content{
    margin: 1em;
}

.profile-pic{
    width: 256px;
    height: 256px;
    border-radius: 4px;
}

.prof-img{
    object-fit: fill;
}

.profile-details{
    display: flex;
    flex-direction: column;
    align-items: center;
}

.profile-action-icons{
    display: flex;
    flex-direction: row;
    /* align-items: center; */
}

.profile-icon-section{
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    margin: 0 0.5em;
    cursor: pointer;
}

.round-circle{
    border-radius: 100%;
    width: 20px;
    height: 20px;
    background: rgb(0, 0, 0, 0.05);
    padding: 1em;
    display: flex;
    justify-content: center;
    align-items: center;
}

.round-circle:hover{
    background: rgb(0, 0, 0, 0.1);
}

.profile-icon-name{
    font-size: 13px;
    color: rgb(0, 0, 0, 0.5);
    margin-top: 0.5em;
}

.personal-section{
    margin-top: 3em;
}

.details-title{
    font-size: 13px;
    color: rgba(0, 0, 0, 0.5);
    margin-top: 1em;
}