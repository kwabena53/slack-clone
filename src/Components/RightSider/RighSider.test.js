import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import RightSider from "./RighSider";

const user = {
  id: 'jbsarfo',
  name: 'Jennifer Sarfo',
  url: 'https://randomuser.me/api/portraits/women/81.jpg',
  email: 'jbsarfo@asu.edu',
  workspaces:['EdPlus at ASU', 'ASU Community'],
  active: true
}

test("renders right sider", () => {
  
  render(<RightSider user={user} />);
  const textElement = screen.getByText(/Jennifer Sarfo/i);
  expect(textElement).toBeInTheDocument();
});

test("matches right sider snapshot", () => {
  const component = renderer.create(<RightSider user={user} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

