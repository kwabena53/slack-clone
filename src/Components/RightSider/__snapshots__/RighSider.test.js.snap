// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches right sider snapshot 1`] = `
<div
  className="rightSider"
>
  <div
    className="contentHeader h-line"
  >
    <h3
      className="slack-h3"
    >
      Profile
    </h3>
    <div
      className="sl-icon"
      onClick={[Function]}
    >
      <svg
        enableBackground="new 0 0 24 24;"
        fill="none"
        height="2em"
        version="1.1"
        viewBox="0 0 24 24"
        width="2em"
        xmlSpace="preserve"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.16998 14.83L14.83 9.17004"
          stroke="black"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1.5"
        />
        <path
          d="M14.83 14.83L9.16998 9.17004"
          stroke="black"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="1.5"
        />
      </svg>
    </div>
  </div>
  <div
    className="rightsider-content"
  >
    <div
      className="profile-details"
    >
      <div>
        <img
          alt="Jennifer Sarfo"
          className="profile-pic"
          src="https://randomuser.me/api/portraits/women/81.jpg"
        />
      </div>
      <div>
        <div
          className="user-name"
        >
          <h3
            className="slack-h3"
          >
            Jennifer Sarfo
          </h3>
        </div>
        <div
          className="online-status"
        />
      </div>
      <div
        className="profile-action-icons"
      >
        <div
          className="profile-icon-section"
        >
          <div
            className="round-circle"
          >
            <svg
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.5 10.5H15.5"
                stroke="black"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeMiterlimit="10"
                strokeWidth="1.5"
              />
              <path
                d="M7 18.4299H11L15.45 21.39C16.11 21.83 17 21.3599 17 20.5599V18.4299C20 18.4299 22 16.4299 22 13.4299V7.42993C22 4.42993 20 2.42993 17 2.42993H7C4 2.42993 2 4.42993 2 7.42993V13.4299C2 16.4299 4 18.4299 7 18.4299Z"
                stroke="black"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeMiterlimit="10"
                strokeWidth="1.5"
              />
            </svg>
          </div>
          <div
            className="profile-icon-name"
          >
            Message
          </div>
        </div>
        <div
          className="profile-icon-section"
        >
          <div
            className="round-circle"
          >
            <svg
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.46005 18.49V15.57C5.46005 14.6 6.22005 13.73 7.30005 13.73C8.27005 13.73 9.14005 14.49 9.14005 15.57V18.38C9.14005 20.33 7.52005 21.9501 5.57005 21.9501C3.62005 21.9501 2.00005 20.32 2.00005 18.38V12.22C1.89005 6.60005 6.33005 2.05005 11.95 2.05005C17.57 2.05005 22 6.60005 22 12.11V18.2701C22 20.2201 20.38 21.84 18.43 21.84C16.48 21.84 14.86 20.2201 14.86 18.2701V15.46C14.86 14.49 15.62 13.62 16.7 13.62C17.67 13.62 18.54 14.38 18.54 15.46V18.49"
                stroke="black"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="1.5"
              />
            </svg>
          </div>
          <div
            className="profile-icon-name"
          >
            Huddle
          </div>
        </div>
        <div
          className="profile-icon-section"
        >
          <div
            className="round-circle"
          >
            <svg
              enableBackground="new 0 0 24 24;"
              fill="black"
              height="1em"
              id="Layer_1"
              stroke="black"
              version="1.1"
              viewBox="0 0 24 24"
              width="1em"
              x="0px"
              xmlSpace="preserve"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
              y="0px"
            >
              <circle
                cx="12"
                cy="3"
                r="1.1"
              />
              <circle
                cx="12"
                cy="12"
                r="1.1"
              />
              <circle
                cx="12"
                cy="21"
                r="1.1"
              />
            </svg>
          </div>
          <div
            className="profile-icon-name"
          >
            More
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="personal-section"
      >
        <div
          className="details-title"
        >
          Display name
        </div>
        <div
          className="details-res"
        >
          jbsarfo
        </div>
      </div>
      <div>
        <div
          className="details-title"
        >
          Local time
        </div>
        <div
          className="details-res"
        >
          6:09 PM
        </div>
      </div>
      <div>
        <div
          className="details-title"
        >
          Email address
        </div>
        <div
          className="details-res"
        >
          jbsarfo@asu.edu
        </div>
      </div>
      <div>
        <div
          className="details-title"
        >
          Shared Workspaces
        </div>
        <div
          className="details-res"
        >
          EdPlus at ASU
        </div>
      </div>
    </div>
  </div>
</div>
`;
